src/rules-parser.pegjs

```diff
@@ -83,7 +107,7 @@ Schema =
       result.properties = properties.properties;
     }
     if (derived) {
-      result.derivedFrom = derived[2];
+      result.derivedFrom = ensureUpperCase(derived[2], "Type names");
```

Comment:
Yes - with helpful line numbers for each location needing to be changed (editor-friendly).